src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/hmi/rc_on_remote_control_settings_notification.cc

```diff
@@ -93,35 +93,50 @@ void RCOnRemoteControlSettingsNotification::Run() {
     return;
   }
 
-  hmi_apis::Common_RCAccessMode::eType access_mode =
-      hmi_apis::Common_RCAccessMode::INVALID_ENUM;
-  if ((*message_)[app_mngr::strings::msg_params].keyExists(
+  ProcessAccessModeParam();
+  ProcessAllowedParam();
+}
+
+void RCOnRemoteControlSettingsNotification::ProcessAccessModeParam() {
+  if (!(*message_)[app_mngr::strings::msg_params].keyExists(
           message_params::kAccessMode)) {
-    access_mode = static_cast<hmi_apis::Common_RCAccessMode::eType>(
-        (*message_)[app_mngr::strings::msg_params][message_params::kAccessMode]
-            .asUInt());
     SDL_LOG_DEBUG(
-        "Setting up access mode : " << AccessModeToString(access_mode));
-  } else {
-    access_mode = resource_allocation_manager_.GetAccessMode();
-    SDL_LOG_DEBUG("No access mode received. Using last known: "
-                  << AccessModeToString(access_mode));
+        "No access mode received. Using last known: "
+        << AccessModeToString(resource_allocation_manager_.GetAccessMode()));
+    return;
   }
+
+  const auto access_mode = static_cast<hmi_apis::Common_RCAccessMode::eType>(
+      (*message_)[app_mngr::strings::msg_params][message_params::kAccessMode]
+          .asUInt());
+  SDL_LOG_DEBUG("Setting up access mode : " << AccessModeToString(access_mode));
   resource_allocation_manager_.SetAccessMode(access_mode);
+}
+
+void RCOnRemoteControlSettingsNotification::ProcessAllowedParam() {
+  if (!(*message_)[app_mngr::strings::msg_params].keyExists(
+          message_params::kAllowed)) {
+    SDL_LOG_DEBUG("No allowed param received. Using last known: "
+                  << std::boolalpha
+                  << resource_allocation_manager_.is_rc_enabled());
```

Comment:
@AKalinich-Luxoft seems we missed a @briefs for methods:
`is_rc_enabled()` and `set_rc_enabled()`
in resource_allocation_manager.h
It might make sense to add briefs in this PR to make the code more consistent.